build/org.eclipse.cdt.build.gcc.core/src/org/eclipse/cdt/build/gcc/core/GCCToolChain.java

```diff
@@ -649,13 +658,25 @@ public String[] getCompileCommands(ILanguage language) {
 	private Set<String> getResourcesFileExtensions() {
 		IContentTypeManager manager = Platform.getContentTypeManager();
 
-		Set<String> fileExts = new TreeSet<String>(String.CASE_INSENSITIVE_ORDER);
+		Set<String> fileExts = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+		if (manager != null) {
+			IContentType contentTypeCpp = manager.getContentType(CCorePlugin.CONTENT_TYPE_CXXSOURCE);
+			if (contentTypeCpp != null) {
+				fileExts.addAll(Arrays.asList(contentTypeCpp.getFileSpecs(IContentType.FILE_EXTENSION_SPEC)));
+			}
 
-		IContentType contentTypeCpp = manager.getContentType(CCorePlugin.CONTENT_TYPE_CXXSOURCE);
-		fileExts.addAll(Arrays.asList(contentTypeCpp.getFileSpecs(IContentType.FILE_EXTENSION_SPEC)));
+			IContentType contentTypeC = manager.getContentType(CCorePlugin.CONTENT_TYPE_CSOURCE);
+			if (contentTypeC != null) {
+				fileExts.addAll(Arrays.asList(contentTypeC.getFileSpecs(IContentType.FILE_EXTENSION_SPEC)));
+			}
+		}
 
-		IContentType contentTypeC = manager.getContentType(CCorePlugin.CONTENT_TYPE_CSOURCE);
-		fileExts.addAll(Arrays.asList(contentTypeC.getFileSpecs(IContentType.FILE_EXTENSION_SPEC)));
+		if (fileExts.isEmpty()) {
+			Activator.log(new Status(Status.ERROR, Activator.getId(),
+					"Error: cannot determine resources file extensions.\n"));
+			fileExts.addAll(Arrays.asList(new String[] { "C", "c", "c++", "cc", "cpp", "cxx" })); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$ //$NON-NLS-6$
```

Comment:
I'm a bit torn here. I don't know if we should hard code the file extensions like this or if we should return the empty set or perhaps throw some kind of exception instead.
@jonahgraham: What's your opinion here?

Reply:
The set is filled at the creation of GCCToolChain. An empty set disables the C/C++ indexing, because no resources will be found in the command lines. An exception will stop the creation of GCCToolChain which will disrupt the whole Core Build Makefile flow.

Reply:
> @jonahgraham: What's your opinion here?

I don't understand the use case that leads to fileExts being empty at this point in the code. Is this a theoretical problem or is there a use case? If it is theoretical I would probably do it as @ewaterlander has it now as it seems a fairly safe fallback way to fail.

Reply:
> I don't understand the use case that leads to fileExts being empty at this point in the code. Is this a theoretical problem or is there a use case? If it is theoretical I would probably do it as @ewaterlander has it now as it seems a fairly safe fallback way to fail.

I'm building a plugin based on Core Build Makefile projects. I'm running static code analysis on my own files and a few CDT files that I needed to modify. I'm currently stuck on CDT 10.5. I created the fallback to get rid of a null pointer issue, because code has to be clean. So you can say it is theoretical. When I move to CDT 11.2 which includes all my other changes I don't need the modified GCCToolChain anymore. But due to constraints I can't move to CDT 11.x for the time being. I offered this change as a code cleanup. I do not depend on it. So you can also choose to remove the fallback. Or I can do it if you want.